switchlink/switchlink_main.c

```diff
@@ -288,6 +289,13 @@ static void nl_process_event_loop(void) {
 
     ret = select(num_fds, &read_fds, NULL, NULL, NULL);
     if (ret == -1) {
+      // Select system call being interrupted by a signal EINTR.
```

Comment:
Good comment. You explain _why_ you are making this test.

Reply:
During automation we saw this issue, where intermittently switchlink_main thread is exited and none of the netlink messages are processed by infrap4d.
Upon debug observed that 'Select' system call is being interrupted by EINTR, this causes thread to exit.

While reading about this issue, figure out this can happen to any Blocking call. In our case we dont have any timeout for Select system call. Hence the chances of getting this interrupt is high.